catroid/src/main/java/org/catrobat/catroid/content/actions/GoNStepsBackAction.java

```diff
@@ -79,14 +72,13 @@ private void goNStepsBack(int steps) {
 		List<Sprite> spriteList = ProjectManager.getInstance().getCurrentProject().getSpriteListWithClones();
 
 		for (int i = 0; i < spriteList.size(); i++) {
-			if (steps > 0) {
-				if (spriteList.get(i).look.getZIndex() >= newSpriteZIndex && spriteList.get(i).look.getZIndex() < zPosition) {
-					spriteList.get(i).look.setZIndex(spriteList.get(i).look.getZIndex() + 1);
-				}
-			} else {
-				if (spriteList.get(i).look.getZIndex() <= newSpriteZIndex && spriteList.get(i).look.getZIndex() > zPosition) {
-					spriteList.get(i).look.setZIndex(spriteList.get(i).look.getZIndex() - 1);
-				}
+
+			int currentSpriteZIndex = spriteList.get(i).look.getZIndex();
+
+			if (steps > 0 && currentSpriteZIndex >= newSpriteZIndex && currentSpriteZIndex < zPosition) {
+				spriteList.get(i).look.setZIndex(currentSpriteZIndex + 1);
+			} else if (currentSpriteZIndex <= newSpriteZIndex && currentSpriteZIndex > zPosition) {
+				spriteList.get(i).look.setZIndex(currentSpriteZIndex - 1);
 			}
 		}
```

Comment:
Does this loop really do something useful? When the z-index of a look is changed, all others are automatically updated correctly as well. So it should be sufficient to set the z-index of the current sprite only. Please recheck.